core/src/main/java/org/ironjacamar/core/bootstrapcontext/BootstrapContextCoordinator.java

```diff
@@ -0,0 +1,308 @@
+/*
+ *  IronJacamar, a Java EE Connector Architecture implementation
+ *  Copyright 2016, Red Hat Inc, and individual contributors
+ *  as indicated by the @author tags. See the copyright.txt file in the
+ *  distribution for a full listing of individual contributors.
+ *
+ *  This is free software; you can redistribute it and/or modify it
+ *  under the terms of the Eclipse Public License 1.0 as
+ *  published by the Free Software Foundation.
+ *
+ *  This software is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the Eclipse
+ *  Public License for more details.
+ *
+ *  You should have received a copy of the Eclipse Public License
+ *  along with this software; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ *  02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.ironjacamar.core.bootstrapcontext;
+
+import org.ironjacamar.common.api.metadata.spec.ConfigProperty;
+import org.ironjacamar.core.CoreLogger;
+import org.ironjacamar.core.workmanager.WorkManagerCoordinator;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.jboss.logging.Logger;
+
+/**
+ * Coordinator for BootstrapContext instances
+ * @author <a href="jesper.pedersen@ironjacamar.org">Jesper Pedersen</a>
+ */
+public class BootstrapContextCoordinator
+{
```

Comment:
The coordinator should use the private API - which could be called CloneableBootstrapContext